test/e2e/framework/test_context.go

```diff
@@ -342,28 +342,28 @@ func RegisterClusterFlags(flags *flag.FlagSet) {
 	flags.StringVar(&TestContext.Tooling, "tooling", "", "The tooling in use (kops, gke, etc.)")
 	flags.StringVar(&TestContext.OutputDir, "e2e-output-dir", "/tmp", "Output directory for interesting/useful test data, like performance data, benchmarks, and other metrics.")
 	flags.StringVar(&TestContext.Prefix, "prefix", "e2e", "A prefix to be added to cloud resources created during testing.")
-	flags.StringVar(&TestContext.MasterOSDistro, "master-os-distro", "debian", "The OS distribution of cluster master (debian, ubuntu, gci, coreos, or custom).")
+	flags.StringVar(&TestContext.MasterOSDistro, "control-plane-os-distro", "debian", "The OS distribution of cluster control plane (debian, ubuntu, gci, coreos, or custom).")
```

Comment:
If you rename a public facing API like a command line flag we also need to have a plan for updating callers setting that flag and it probably needs a release note for callers we do not control.

Reply:
Renaming these makes sense, but we need to take some consideration in how we go about enacting it to not just break CI. One approach would be roughly:
- add two flags for some time
- get CI migrated
- have a release note about the old name being deprecated
- having it print a warning if the old name is used
- in some future release deleting the old flag

breaking changes with this many steps usually have a KEP so we've documented the plan somewhere. https://github.com/kubernetes/kubernetes/issues/103144#issuecomment-876826747

Reply:
I think it probably makes the most sense to leave flags/environment variables to a different PR so the other more easily merged changes can merge quicker while someone scopes out the external / breaking changes